contracts/proxy/diamond/writable/DiamondWritableInternal.sol

```diff
@@ -95,123 +103,124 @@ abstract contract DiamondWritableInternal is IDiamondWritableInternal {
-                    selectorInSlotIndex--;
+                    lastSelectorIndexInSlug--;
                 }
 
-                bytes4 lastSelector;
-                uint256 oldSelectorsSlotCount;
-                uint256 oldSelectorInSlotPosition;
-
-                // adding a block here prevents stack too deep error
-                {
-                    // replace selector with last selector in l.facets
-                    lastSelector = bytes4(
-                        selectorSlot << (selectorInSlotIndex << 5)
-                    );
-
-                    if (lastSelector != selector) {
-                        // update last selector slot position info
-                        l.facets[lastSelector] =
-                            (oldFacet & CLEAR_ADDRESS_MASK) |
-                            bytes20(l.facets[lastSelector]);
-                    }
-
-                    delete l.facets[selector];
-                    uint256 oldSelectorCount = uint16(uint256(oldFacet));
-                    oldSelectorsSlotCount = oldSelectorCount >> 3;
-                    oldSelectorInSlotPosition = (oldSelectorCount & 7) << 5;
+                // extract the last selector from the last slug
+                // it will be used to overwrite the selector being removed
+                bytes4 lastSelector = bytes4(
+                    lastSlug << (lastSelectorIndexInSlug << 5)
+                );
+
+                if (lastSelector != selector) {
+                    // update last slug position info
+                    l.selectorInfo[lastSelector] =
+                        (selectorInfo & CLEAR_ADDRESS_MASK) |
+                        bytes20(l.selectorInfo[lastSelector]);
                 }
 
-                if (oldSelectorsSlotCount != selectorSlotCount) {
-                    bytes32 oldSelectorSlot = l.selectorSlots[
-                        oldSelectorsSlotCount
-                    ];
+                uint256 slugIndex = uint16(uint256(selectorInfo)) >> 3;
+                uint256 selectorBitIndexInSlug = (uint16(
+                    uint256(selectorInfo)
+                ) & 7) << 5;
 
-                    // clears the selector we are deleting and puts the last selector in its place.
-                    oldSelectorSlot =
-                        (oldSelectorSlot &
-                            ~(CLEAR_SELECTOR_MASK >>
-                                oldSelectorInSlotPosition)) |
-                        (bytes32(lastSelector) >> oldSelectorInSlotPosition);
+                // overwrite the selector being deleted with the last selector in the array
 
-                    // update storage with the modified slot
-                    l.selectorSlots[oldSelectorsSlotCount] = oldSelectorSlot;
+                if (slugIndex == lastSlugIndex) {
+                    // selector is being removed from the last slug, which has already been loaded
+                    // slug needs not be written to storage because it continues to be tracked
```

Comment:
I think this comment should be made clearer

Reply:
Would this be resolved by the function name change you suggested below?

Reply:
```

// the selector being removed is from the last slug, this means we replace it in place
// the slug will be written to storage upstream, when returned. 
```

Reply:
Made some changes here.  There's also another comment above the if/else blocks that applies to both cases.